crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3645,6 +3645,25 @@ pub struct Class<'db> {
     known: Option<KnownClass>,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+enum InheritanceCycle {
+    // The class is cyclically defined and is a participant in the cycle.
+    // i.e., it inherits either directly or indirectly from itself.
+    Participant,
+    // The class inherits from a class that is a `Participant` in an inheritance cycle,
+    // but is not itself a participant.
```

Comment:
Using `///` over `//` has the advantage that the text shows up as documentation. It's otherwise considered an inline comment.

```suggestion
    /// The class is cyclically defined and is a participant in the cycle.
    /// i.e., it inherits either directly or indirectly from itself.
    Participant,
    /// The class inherits from a class that is a `Participant` in an inheritance cycle,
    /// but is not itself a participant.
```